### PR TITLE
Add CI to build and test null and DML backend.

### PR DESCRIPTION
**Pull request by BruceDai**
@huningxin PTAL, thanks.

### CODE REVIEW

**Comment by BruceDai** [comment]
The status badges like:
null backend
DirectML backend

**Review by huningxin — COMMENTED** [review]
The CI would be very helpful, thanks much @BruceDai !

Did you consider adding the coding style check and other backends support, like OpenVINO?

**Comment by BruceDai** [comment]
> Did you consider adding the coding style check and other backends support, like OpenVINO?

Yes, I'm ongoing to adding the coding style check and OpenVINO backend, thanks.

**Comment by huningxin** [comment]
> Yes, I'm ongoing to add the coding style check and OpenVINO backend, thanks.

Please file corresponding issues and we merge this PR.

**Comment by BruceDai** [comment]
> Please file corresponding issues and we merge this PR.

Thanks @huningxin, I filed https://github.com/webmachinelearning/webnn-native/issues/31.

**Review by huningxin — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -81,6 +81,11 @@ Run end2end tests, for example `./out/Release/webnn_end2end_tests`.
  * The current implementation of XNNPACK and oneDNN backends is mainly for the investigation of WebNN [Operation Level Execution
 ](https://webmachinelearning.github.io/webnn/#usecase-op-level-exec) use case. So only a limited set of tests (such as of conv2d) is expected to pass.
 
+### Build and Run tests Status
```
Comment (huningxin) [inline README.md:84]: @BruceDai, although this PR has been merged, could you please send another PR to move the build status to the head as https://github.com/webmachinelearning/webnn-polyfill does. Thanks.

File: workflow_scripts/src/ci.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+      }
+    });
+  }
+};
+
+(async () => {
+  await installDepotTools();
+  await syncCode();
+  await buildTest();
+})().catch((error) => console.error(error));
```
Comment (huningxin) [inline workflow_scripts/src/ci.js:148]: According to https://github.com/webmachinelearning/webnn-native/runs/3073757456?check_suite_focus=true, there are 1 test case failed
```shell
[----------] Global test environment tear-down
[==========] 172 tests from 25 test suites ran. (87337 ms total)
[  PASSED  ] 171 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] ReduceMeanTests.ReduceMeanDefault

 1 FAILED TEST
  YOU HAVE 3 DISABLED TESTS
```

However, the build and test still passed. I think it should report error for the failures. What do you think?

File: README.md (hunk 1 above)
Comment (BruceDai) [reply README.md:84]: Fixed by https://github.com/webmachinelearning/webnn-native/pull/32, thanks.

File: workflow_scripts/src/ci.js (hunk 2 above)
Comment (BruceDai) [reply workflow_scripts/src/ci.js:148]: > However, the build and test still passed. I think it should report error for the failures. What do you think?

Do you mean that CI should fail on this case?
And I think CI should pass with build latest code successfully and none regression test which I'm working to implement.

Comment (huningxin) [reply workflow_scripts/src/ci.js:148]: > And I think CI should pass with build latest code successfully and none regression test which I'm working to implement.

It makes sense. Let's review once you have PR for regression check.

However, I think we should have tracking issue for each failed test case. What do you think?

Comment (BruceDai) [reply workflow_scripts/src/ci.js:148]: > However, I think we should have tracking issue for each failed test case. What do you think?

Do you mean tracking issue on github issues? The failed ReduceMeanTests.ReduceMeanDefault test was reported https://github.com/webmachinelearning/webnn-native/issues/19.

Comment (huningxin) [reply workflow_scripts/src/ci.js:148]: That's a good one. Thanks for the pointer.
